### PR TITLE
Maven cleanup, use WorldGuardWrapper, fix region checks

### PR DESCRIPTION
**Pull request by sgdc3**

### CODE REVIEW

File: plugin/src/main/java/net/mcjukebox/plugin/bukkit/listeners/RegionListener.java [hunk 1]
```diff
@@ -8,55 +8,42 @@
...
-
-    @EventHandler
-    public void onMove(PlayerMoveEvent e){
+    private void handleMovement(Player player, Location from, Location to) {
         //Only execute if the player moves an entire block
-        if(!(e.getFrom().getX() != e.getTo().getX() || e.getFrom().getZ() != e.getTo().getZ())) return;
+        if (from != null
+                && from.getBlockX() == to.getBlockX()
+                && from.getBlockY() == to.getBlockY()
+                && from.getBlockZ() != to.getBlockZ()) return;
```
Comment (Moutard3) [inline plugin/src/main/java/net/mcjukebox/plugin/bukkit/listeners/RegionListener.java]: Shouldn't it be like this?
```suggestion
        if (from != null
                && from.getBlockX() == to.getBlockX()
                && from.getBlockY() != to.getBlockY()
                && from.getBlockZ() == to.getBlockZ()) return;
```

**Review by Moutard3 — COMMENTED** [review]
Looks great!

We may want to edit ci as it tries to run installSkript

File: plugin/src/main/java/net/mcjukebox/plugin/bukkit/listeners/RegionListener.java (hunk 1 above)
Comment (sgdc3) [reply plugin/src/main/java/net/mcjukebox/plugin/bukkit/listeners/RegionListener.java]: Oh god, i tried to fix broken logic with another broken logic ahahahah

Comment (sgdc3) [reply plugin/src/main/java/net/mcjukebox/plugin/bukkit/listeners/RegionListener.java]: It should be == in each axis :)

Comment (sgdc3) [reply plugin/src/main/java/net/mcjukebox/plugin/bukkit/listeners/RegionListener.java]: and no, we NEED to check movements on the y axis as regions can cohexist at different Ys

**Comment by sgdc3** [comment]
here we go! :)

File: plugin/src/main/java/net/mcjukebox/plugin/bukkit/listeners/RegionListener.java (hunk 1 above)
Comment (Moutard3) [reply plugin/src/main/java/net/mcjukebox/plugin/bukkit/listeners/RegionListener.java]: Yep that's true, then == everywhere 👍

**Comment by Moutard3** [comment]
Great! Now you need to wait feedback from @oliverdunk ;)

**Comment by oliverdunk** [comment]
> Great! Now you need to wait feedback from @oliverdunk ;)

Thanks for taking a look at this @Moutard3.

@sgdc3, thanks for your contribution! The fixes and refactoring here are great. I'm curious if there's a particular WorldGuard version you're introducing WorldGuardWrapper to support, or if you just thought it was a nice fit (it looks like you work on it)? I'm not against it, but our current solution is working well, and would allow us to support other plugins in the future.

Feel free to reply here. Splitting your MR up might be nice, so we can merge the other fixes sooner. Not essential though.

**Comment by sgdc3** [comment]
@oliverdunk WGW won't stop you from adding hooks to other protection plugins, it just allows you to delegate the cross version support no-sense to a more mature cross-project library. 😃

**Comment by oliverdunk** [comment]
@sgdc3 I understand this, but our current approach is working well. While your project may be slightly older, our implementation has been around for a while too.

At the moment, I don't plan on merging this. I'd be happy to accept a PR with the other fixes, but I don't think this is the right time to adopt that library.

Going forward, please don't hesitate to reach out on Discord, or open an issue, before making a change. I appreciate the time you spent here, and would be happy to chat in the future so you don't work on something that doesn't end up getting accepted.

**Comment by sgdc3** [comment]
I have no time to rework this PR atm sorry, i'll continue to use my fork as the current official release still has huge performance issues due to that PlayerMoveEvent listener logic error.

**Comment by oliverdunk** [comment]
> I have no time to rework this PR atm sorry, i'll continue to use my fork as the current official release still has huge performance issues due to that PlayerMoveEvent listener logic error.

No worries, that's totally fine. It is a bad bug so I hope to release a fix with it soon.

**Pull request closed without merging**